source/main.c

```diff
@@ -47,15 +51,17 @@ void dol_alloc(int size)
         return;
     }
 
-    dol = (u8 *) memalign(32, size);
+    u8 *dol = (u8 *) memalign(32, size);
 
     if (!dol)
     {
         kprintf("Couldn't allocate memory\n");
     }
+
+    *_dol = dol;
 }
 
-void load_parse_cli(char *path)
+void load_parse_cli(char **_dol_argv, int *_dol_argc, char *path)
```

Comment:
Instead of the first two arguments here, take in a pointer to a `struct boot_payload`.